frontend/src/app/search.tsx

```diff
@@ -46,44 +54,63 @@ export default class Search extends Component {
+                            <span className="sp-prop" id="span-upload-time">Upload Time</span>
+                          </DataListCell>,]} />
                 </DataListItemRow>
                 {this.state.data.map(data => (
-                  <DataListItemRow id="data-rows" key={data["pant:transientPath"]} onClick={() => this.setPreview(data["pant:transientPath"])}>
+                  <DataListItemRow id="data-rows">
                     <DataListItemCells
-                      dataListCells={[
-                        <DataListCell width={4} key="title">
-                          {data["jcr:title"]}
-                        </DataListCell>,
-                        <DataListCell width={4} key="description">
-                          {data["jcr:description"]}
-                        </DataListCell>,
-                        <DataListCell key="resource source">
-                          {data["pant:transientSource"]}
-                        </DataListCell>,
-                        <DataListCell key="source name">
-                          {data["pant:transientSourceName"]}
-                        </DataListCell>,
-                        <DataListCell key="upload time">
-                          {this.formatDate(new Date(data["jcr:created"]))}
-                        </DataListCell>,]} />
+                          dataListCells={[
+                          <DataListCell width={1} key="checkbox">
+                            <Checkbox
+                                isChecked={this.state.check}
+                                onChange={this.handleChange}
+                                aria-label="controlled checkbox example"
+                                id="check"
+                                name={data["pant:transientPath"]}
+                            />
+                          </DataListCell>,    
+                          <DataListCell width={4} key={data["pant:transientPath"]} onClick={() => this.setPreview(data["pant:transientPath"])}>
+                            {data["jcr:title"]}
+                          </DataListCell>,
+                          <DataListCell width={4} key={data["pant:transientPath"]} onClick={() => this.setPreview(data["pant:transientPath"])}>
+                            {data["jcr:description"]}
+                          </DataListCell>,
+                          <DataListCell key={data["pant:transientPath"]} onClick={() => this.setPreview(data["pant:transientPath"])}>
+                            {data["pant:transientSource"]}
+                          </DataListCell>,
+                          <DataListCell key={data["pant:transientPath"]} onClick={() => this.setPreview(data["pant:transientPath"])}>
+                            {data["pant:transientSourceName"]}
+                          </DataListCell>,
+                          <DataListCell key={data["pant:transientPath"]} onClick={() => this.setPreview(data["pant:transientPath"])}>
```

Comment:
There is probably a way for us to not have to repeat `onClick={() => this.setPreview(data["pant:transientPath"])}` 5 times. Please find a way to pull that out into a common method.

Reply:
Acknowledged.
I will investigate on this and will work on simplifying it and putting it into a common set of cells., so that the onclick() function could be called just once.